### PR TITLE
Fix cppcheck 1.87 warnings in Crystal

### PR DESCRIPTION
**Pull request by soininen**
This PR fixes cppcheck 1.87 warnings in the Crystal module.

**To test:**

Code review.

Fixes partially #22912.

*This does not require release notes* because this is an internal change.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/Crystal/src/ConnectedComponentLabeling.cpp [hunk 1]
```diff
@@ -439,12 +439,11 @@ ClusterTuple ConnectedComponentLabeling::executeAndFetchClusters(
...
   VecIndexes keys;
   keys.reserve(clusters.size());
-  for (auto &cluster : clusters) {
-    keys.push_back(cluster.first);
-  }
+  std::transform(clusters.cbegin(), clusters.cend(), std::back_inserter(keys),
+                 [](const auto &cluster) { return cluster.first; });
   // Write each cluster out to the output workspace
   PARALLEL_FOR_NO_WSP_CHECK()
-  for (int i = 0; i < static_cast<int>(keys.size()); ++i) { // NOLINT
```
Comment (TTitcombe) [inline Framework/Crystal/src/ConnectedComponentLabeling.cpp]: I've been doing clang-tidy modernize-loop-convert and compilation seems to fail if you use a range-based loop after a PARALLEL check, which is probably why the // NOLINT was put in. Is // NOLINT tripping up cppcheck or is there a Mantid standards reason for removing it?

File: Framework/Crystal/src/IntegratePeakTimeSlices.cpp [hunk 2]
```diff
@@ -427,7 +422,7 @@ void IntegratePeakTimeSlices::exec() {
           if (m_R0 > 0)
             Radius = m_R0;
 
-          int NN = m_NeighborIDs[1];
+          const int NN = m_NeighborIDs[1];
```
Comment (TTitcombe) [inline Framework/Crystal/src/IntegratePeakTimeSlices.cpp]: Can we change this to `neighbourId` as well to bring it inline with the change below? Or if we need to have separate variables can we use a name which explains why we need two `neighbourId`s?

**Review by TTitcombe — CHANGES_REQUESTED** [review]

File: Framework/Crystal/src/ConnectedComponentLabeling.cpp (hunk 1 above)
Comment (soininen) [reply Framework/Crystal/src/ConnectedComponentLabeling.cpp]: `cppcheck` ignores such comments. I was deleting stuff too hastily here. I'll put the comment back.

File: Framework/Crystal/src/IntegratePeakTimeSlices.cpp (hunk 2 above)
Comment (soininen) [reply Framework/Crystal/src/IntegratePeakTimeSlices.cpp]: I'll remove both temporary variables to solve the issue. That seems the most straightforward thing to do as I don't want to dive into the code to invent more spot-on names.

**Review by TTitcombe — DISMISSED** [review]

**Review by DanNixon — CHANGES_REQUESTED** [review]

**Comment by soininen** [comment]
I am sorry for force-pushing. I did not want to leave a broken build in the history.

**Comment by DanNixon** [comment]
@soininen Fine by me, that's what the ability to do so is there for (ot at least one things it's there for).

**Review by DanNixon — APPROVED** [review]

**Pull request merged**